### DISCUSSION TITLE
Data buckets generation

### DISCUSSION THREAD

**Discussion by jcbenck** [mockoon/mockoon#1240] [Help] [answered]
Is it possible to regenerate a Data Bucket without having to restart the server? I'm encountering an issue where updates made to a 'parent' Data Bucket are not being reflected in a 'child' Data Bucket that references it. How can I ensure that changes in the parent Data Bucket are updated in the child Data Bucket in real-time or without server restart?

**Comment by 255kb** [comment, marked as answer]
This is currently not possible. Data buckets are evaluated during startup, eventually using data from another bucket which was generated **before**, but after this, the template helpers disappear. 

I already thought about something like this, but I have no precise idea on how it would look: 
- An option to generate the bucket on each request, like a regular body. But then why not use a regular body? As a wordaround I would do this. Use a regular body which are regenerated on each request.
- A specific PURGE endpoint for CRUD routes to regenerate the body. But what if you are not using the CRUD routes? 
- A new helper used to wrap other helpers and indicate they should be evaluated at runtime only: `{{ runtime (helperName) }}`. We already have this mechanism with [request helpers that are evaluated during the first call](https://mockoon.com/docs/latest/data-buckets/overview/#data-buckets-generation).

**Reply by jcbenck** [reply]
Thanks for the answer @255kb. For me the approach of helpers that indicate that the value should be evaluated at runtime seems the best.
